packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -719,9 +727,9 @@ class _DraggableScrollableSheetState extends State<DraggableScrollableSheet> {
     _scrollController.extent = _extent;
     // If an external facing controller was provided, let it know that the
     // extent has been replaced.
-    if (widget.controller == oldWidget.controller) {
-      widget.controller?._onExtentReplaced(previousExtent);
-    }
+    widget.controller?._onExtentReplaced(previousExtent);
+    previousExtent.dispose();
```

Comment:
Without this may cause some memory leaks.

Reply:
Nice